keras_nlp/layers/modeling/transformer_decoder.py

```diff
@@ -370,13 +370,17 @@ def call(
         residual = x
         if self.normalize_first:
             x = self._self_attention_layer_norm(x)
-        x, self_attention_cache = self._self_attention_layer(
+        attention_output = self._self_attention_layer(
             query=x,
             value=x,
             attention_mask=self_attention_mask,
             cache=self_attention_cache,
             cache_update_index=self_attention_cache_update_index,
         )
+        if self_attention_cache is None:
```

Comment:
This seems like a potentially meaningful behavior change -- is it intended to be part of this PR?

Reply:
Ah see comment above. Basically our cached attention misbehaves during a symbolic call. Adding this test suite just exposed that.

https://github.com/keras-team/keras-nlp/pull/1238/files/b0163bd25341dbc978faf1db5c62785100cf7bf1#diff-0fb2c02147a044e77aab14f23846e1d821fdf556038ba75cde583192704960a5

Apparently returning `None` from a layer is a no-no, which makes optional return types a little more annoying. I could also disable the test for cached attention and fix separately, but that would blow up the number of PRs I need review for.